app/controllers/posts_controller.rb

```diff
@@ -1,72 +1,5 @@
 class PostsController < ApplicationController
-
-
-  def show
-     @topic = Topic.find(params[:topic_id])
-     @post = Post.find(params[:id])
-     @comment = Comment.new
-     @comments = @post.comments 
-     authorize @topic 
+  def index
+    @posts = Post.visible_to(current_user).where("Posts.created_at > ?", 7.days.ago).paginate(page: params[:page], per_page: 10)
```

Comment:
you should write `posts.created_at` instead of `Posts.created_at` if your table name is `posts`. It may work with sqlite but not with all dbms.